app/scripts/components/exploration/components/timeline/timeline-controls.tsx

```diff
@@ -255,26 +304,100 @@ const getCalendarView = (timeDensity: TimeDensity): View => {
 };
 
 export function TimelineControls(props: TimelineControlsProps) {
-  const { xScaled, width, outOfViewHeads, onZoom, timeDensity, timelineLabelsFormat, minMaxTemporalExtent } = props;
+  const {
+    xScaled,
+    width,
+    outOfViewHeads,
+    onZoom,
+    timeDensity,
+    timelineLabelsFormat,
+    minMaxTemporalExtent
+  } = props;
 
   const [selectedDay, setSelectedDay] = useAtom(selectedDateAtom);
+
   const [selectedCompareDay, setSelectedCompareDay] = useAtom(
     selectedCompareDateAtom
   );
   const [selectedInterval, setSelectedInterval] = useAtom(selectedIntervalAtom);
   const { features } = useAois();
 
   // Scale to use when there are no datasets with data (loading or error)
-  const initialScale = useMemo(() => getInitialScale(width) ,[width]);
+  const initialScale = useMemo(() => getInitialScale(width), [width]);
 
-  const calendarView = useMemo(() => getCalendarView(timeDensity), [timeDensity]);
+  const calendarView = useMemo(
+    () => getCalendarView(timeDensity),
+    [timeDensity]
+  );
+  //Center to selected point
+  const { onTOIZoom } = useOnTOIZoom();
+  const timelineWidth = useAtomValue(timelineWidthAtom);
+  const { k } = useAtomValue(zoomTransformAtom);
+
+  const { main } = useScales();
+  const centerTimelineOnSelections = useCallback(
+    (newDate) => {
+      if (!timelineWidth || !main) return;
+
+      //setting most recent date value depending on interaction
+      const newSelectedDay = newDate.selectedDay ?? selectedDay;
+      const newSelectedCompareDay =
+        newDate.selectedCompareDay ?? selectedCompareDay;
+      const newSelectedStartInterval = newDate.start ?? selectedInterval?.start;
+      const newSelectedEndInterval = newDate.end ?? selectedInterval?.end;
+
+      //Settign visible areas to base calculations
+      const visibleTimeline =
+        timelineWidth - RIGHT_AXIS_SPACE - HEADER_COLUMN_WIDTH;
+      const widthToFit = visibleTimeline * 0.9;
+      const timelineCenter = visibleTimeline * 0.5;
+      const startPoint = 0;
+      let new_k;
+      let new_x;
+      if (newSelectedDay) {
+        new_k = k;
+        new_x = startPoint - new_k * main(newSelectedDay) + timelineCenter;
+      }
+      if (newSelectedCompareDay) {
+        if (newSelectedDay < newSelectedCompareDay) {
+          new_k =
+            widthToFit / (main(newSelectedCompareDay) - main(newSelectedDay));
+          new_x = startPoint - new_k * main(newSelectedDay);
+        } else {
+          new_k =
+            widthToFit / (main(newSelectedDay) - main(newSelectedCompareDay));
+          new_x = startPoint - new_k * main(newSelectedCompareDay);
+        }
+      }
+
+      if (newSelectedStartInterval && newSelectedEndInterval) {
+        new_k =
+          widthToFit /
+          (main(newSelectedEndInterval) - main(newSelectedStartInterval));
+        new_x = startPoint - new_k * main(newSelectedStartInterval);
+      }
```

Comment:
There's repetition in how `new_k` and `new_x` are calculated in the code block below, the only difference is the specific days being compared. It might be helpful to create a utility function that handles the calculations of `new_x` and `new_k` and reuse it in both blocks to simplify the code

```

if (newSelectedCompareDay) {
        if (newSelectedDay < newSelectedCompareDay) {
          new_k =
            widthToFit / (main(newSelectedCompareDay) - main(newSelectedDay));
          new_x = startPoint - new_k * main(newSelectedDay);
        } else {
          new_k =
            widthToFit / (main(newSelectedDay) - main(newSelectedCompareDay));
          new_x = startPoint - new_k * main(newSelectedCompareDay);
        }
      }

      if (newSelectedStartInterval && newSelectedEndInterval) {
        new_k =
          widthToFit /
          (main(newSelectedEndInterval) - main(newSelectedStartInterval));
        new_x = startPoint - new_k * main(newSelectedStartInterval);
      }
```